onnxruntime/core/providers/dml/DmlExecutionProvider/src/GraphDescBuilder.cpp

```diff
@@ -344,35 +372,32 @@ namespace Dml::GraphDescBuilder
-                            // across the graph input as well as every consumer's unique constant node.  However it is currently 
-                            // only used for small inputs.
-                            
-                            // TODO: Rework this to create DML constant nodes with the minimum data size actually used by consuming
-                            // nodes.  This would allow this size to be reduced while handling the case that 1D scale and zero point
-                            // values that have been de-duplicated with conversion to scalars in kernels.
-                            uint32_t c_maxConstNodeDataSize = 1024 * 1024;
+                            auto& operatorGraphInputNode = graphNodeCreateInfo.nodesAsOperatorDesc[operatorGraphInputEdge.ToNodeIndex];
+                            std::vector<DmlBufferTensorDesc*> toNodeInputTensorDescs = operatorGraphInputNode->GetInputTensors();
+                            DmlBufferTensorDesc* tensorDesc = toNodeInputTensorDescs[operatorGraphInputEdge.ToNodeInputIndex];
 
                             ComPtr<OnnxTensorWrapper> constantInput = constantCpuGraphInputGetter(arg->Name());
 
-                            if (constantInput && constantInput->GetTensorByteSize() < c_maxConstNodeDataSize)
+                            if (constantInput)
                             {
                                 std::vector<uint8_t> tensorData;
                                 tensorData.insert(
-                                    tensorData.begin(), 
-                                    static_cast<const uint8_t*>(constantInput->GetData()), 
-                                    static_cast<const uint8_t*>(constantInput->GetData()) + constantInput->GetTensorByteSize());
-                                    
-                                NodeInfo nodeInfo = {};
-                                nodeInfo.nodeDef = std::move(tensorData);
-                                graphNodes.push_back(std::move(nodeInfo));
-
-                                DML_INTERMEDIATE_GRAPH_EDGE_DESC edge = {};
-                                edge.FromNodeIndex = static_cast<UINT>(graphNodes.size() - 1);
-                                edge.FromNodeOutputIndex = 0;
-                                edge.ToNodeIndex = mainGraphNodeIndex;
-                                edge.ToNodeInputIndex = operatorGraphInputEdge.ToNodeInputIndex;
-                                graphIntermediateEdges.push_back(edge);
+                                    tensorData.begin(),
+                                    static_cast<const uint8_t*>(constantInput->GetData()),
+                                    static_cast<const uint8_t*>(constantInput->GetData()) + GetElementSize(tensorDesc->dataType));
+                                if (tensorData.size() <= tensorDesc.TotalTensorSizeInBytes)
+                                {
```

Comment:
This is enabling the first element of the constant to be used as long as the size of that element is <= the total tensor size.  It should be reversed